### PR TITLE
sysroot: Load bootloader configs via boot_fd

### PR DESCRIPTION
**Pull request by cgwalters**
This was a general principle cleanup, preparation
for handling VFAT for /boot for systemd-boot/BLS
support.

However I ran into an ugly corner case in our
unit tests that pointed at a sysroot without a
boot directory. The previous logic handled
ENOENT for boot/loader but not /boot.
Continue to cope with that degenerate situation.

### CODE REVIEW

File: src/libostree/ostree-sysroot.c [hunk 1]
```diff
@@ -338,9 +338,38 @@ ensure_sysroot_fd (OstreeSysroot *self, GError **error)
   return TRUE;
 }
 
+/* Require that both self->sysroot_fd is set.
```
Comment (allisonkarlitskaya) [inline src/libostree/ostree-sysroot.c:341]: Both self->sysroot_fd and... what?

**Review by allisonkarlitskaya — APPROVED** [review]
Looks good, but you've got narrow text width in your commit messages again :)
